packages/grid/src/DataBarCellRenderer.ts

```diff
@@ -0,0 +1,564 @@
+/* eslint-disable class-methods-use-this */
+import { getOrThrow } from '@deephaven/utils';
+import CellRenderer from './CellRenderer';
+import { isExpandableGridModel } from './ExpandableGridModel';
+import { isDataBarGridModel } from './DataBarGridModel';
+import { VisibleIndex } from './GridMetrics';
+import GridColorUtils, { Oklab } from './GridColorUtils';
+import GridUtils from './GridUtils';
+import memoizeClear from './memoizeClear';
+import { GridRenderState } from './GridRendererTypes';
+
+interface DataBarRenderMetrics {
+  /** The total width the entire bar from the min to max value can take up (rightmostPosition - leftmostPosition) */
+  maxWidth: number;
+  /** The x coordinate of the bar (the left) */
+  x: number;
+  /** The y coordinate of the bar (the top) */
+  y: number;
+  /** The position of the zero line */
+  zeroPosition: number;
+  /** The position of the leftmost point */
+  leftmostPosition: number;
+  /** The position of the rightmost point */
+  rightmostPosition: number;
+  /** The range of values (e.g. max of 100 and min of -50 means range of 150) */
+  totalValueRange: number;
+  /** The width of the databar */
+  dataBarWidth: number;
+  /** The x coordinates of the markers (the left) */
+  markerXs: number[];
+}
+class DataBarCellRenderer extends CellRenderer {
+  drawCellContent(
+    context: CanvasRenderingContext2D,
+    state: GridRenderState,
+    column: VisibleIndex,
+    row: VisibleIndex
+  ) {
+    const { metrics, model, theme } = state;
+    if (!isDataBarGridModel(model)) {
+      return;
+    }
+    const {
+      modelColumns,
+      modelRows,
+      allRowHeights,
+      allRowYs,
+      firstColumn,
+    } = metrics;
+
+    const isFirstColumn = column === firstColumn;
+    const rowHeight = getOrThrow(allRowHeights, row);
+    const modelRow = getOrThrow(modelRows, row);
+    const modelColumn = getOrThrow(modelColumns, column);
+    const rowY = getOrThrow(allRowYs, row);
+    const textAlign = model.textAlignForCell(modelColumn, modelRow);
+    const text = model.textForCell(modelColumn, modelRow);
+    const value = Number(text);
+    const { x: textX, y: textY } = GridUtils.getTextRenderMetrics(
```

Comment:
I think the text is drawing a little bit high - it doesn't appear to be vertically centered in the cell. I have not debugged why that is.

Reply:
Not sure why this works but this does seem to solve the problem, however it is like a hacky solution: [https://stackoverflow.com/a/63239887/20005358](https://stackoverflow.com/a/63239887/20005358)

Reply:
Hmm. It looks like the vertical position is actually the same as it is currently...
I'll defer to @dsmmcken for this, but I think for the data bars in particular I'd bump them down a couple of pixels, just to vertically center it. Not sure if our regular text cell rendering is off as well.

Did a bit of debugging by drawing a red line where the `textX`/`textY` is:
![image](https://user-images.githubusercontent.com/4505624/233650865-23549678-07cf-4437-8284-301625811a08.png)

Did this just be adding the following in:
```
    context.strokeStyle = 'red';
    context.beginPath();
    context.moveTo(textX - 50, textY);
    context.lineTo(textX + 50, textY);
    context.stroke();
    context.closePath();
```

So we're using the `textBaseline = 'middle'`, which draws the text with the middle: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textBaseline
That includes descenders, but numbers don't ever descend past the baseline, which is possibly why it looks so high/off.

The [example you linked](https://stackoverflow.com/a/63239887/20005358) is using the bounding box descent, which we don't really care about (since we know numbers don't descend).

What we could do instead is set `textBaseline = 'bottom';`, then we position the text to vertically center it; we're using a monospace font so all numbers should be the same height and we only need to measure once (i.e. just measure `'1234567890'`) and can cache that result.
Then we can position it within the cell on the bottom baseline, so something like `cellTextY = (rowHeight - textHeight) * 0.5`